### PR TITLE
Add docs for a part of the project

### PR DESCRIPTION
**Pull request by nbirillo**
Add docs for several parts of the compiler plugin:
- common utils
- scripting
- entry points of the plugin

Make small code refactoring

### CODE REVIEW

**Review by elena-lyulina — CHANGES_REQUESTED** [review]
Thanks :)
I tried to fix some English, but there is still a couple of comments which i don't totally understand, could you help?

File: reflekt-plugin/src/main/kotlin/org/jetbrains/reflekt/plugin/utils/Keys.kt [hunk 1]
```diff
@@ -6,6 +6,21 @@ import org.jetbrains.kotlin.config.CompilerConfigurationKey
...
 import java.io.File
 
+/**
+ * The Kotlin compiler configuration keys for the Reflekt plugin
+ *
+ * @property OUTPUT_DIR stores a relative path for generated files (e.g. ReflektImpl.kt)
+ * @property ENABLED indicates if the plugin is enabled
+ * @property DEPENDENCY_JARS stores the libraries jars absolute path that included in the current project
+ *  as a compileClasspath configuration
+ * @property REFLEKT_META_FILES stores the relative path to the ReflektMeta file
```
[the remark this answers is not in the document]
Comment (nbirillo) [reply reflekt-plugin/src/main/kotlin/org/jetbrains/reflekt/plugin/utils/Keys.kt]: Sorry, for REFLEKT_META_PATH everything is correct, for REFLEKT_META_FILES - stores the absolute file's path from the library with Reflekt meta information

File: reflekt-plugin/src/main/kotlin/org/jetbrains/reflekt/plugin/utils/PluginConfig.kt [hunk 2]
```diff
@@ -9,9 +9,27 @@ import org.jetbrains.kotlin.config.CompilerConfiguration
...
 
+/**
+ * A class to parse and store the command line arguments from the plugin
+ *
+ * @param configuration the current Kotlin compiler configuration or null
+ * @param logFilePath path to the file with logs. By default is pathToKotlin/daemon/reflekt-log.log
+ * @param isTestConfiguration indicates if the plugin is used in tests
+ *
+ * @property enabled indicates if the plugin is enabled
+ * @property reflektMetaFilesFromLibraries stores the relative path to the ReflektMeta file
```
[the remark this answers is not in the document]
Comment (nbirillo) [reply reflekt-plugin/src/main/kotlin/org/jetbrains/reflekt/plugin/utils/PluginConfig.kt:20]: the same errors( reflektMetaFilesFromLibraries stores the absolute file's path from the library with Reflekt meta information

File: reflekt-plugin/src/main/kotlin/org/jetbrains/reflekt/plugin/utils/compiler/KotlinLightVirtualFile.kt [hunk 3]
```diff
@@ -4,11 +4,17 @@ import org.jetbrains.kotlin.com.intellij.testFramework.LightVirtualFile
 import org.jetbrains.kotlin.idea.KotlinLanguage
 import java.io.File
 
-/** Wrapper for VirtualFile that retains path on machine */
+/**
+ * Wrapper for [LightVirtualFile] that retains path on machine
```
[the remark this answers is not in the document]
Comment (nbirillo) [reply reflekt-plugin/src/main/kotlin/org/jetbrains/reflekt/plugin/utils/compiler/KotlinLightVirtualFile.kt:8]: of the current user

**Comment by nbirillo** [comment]
Thank you! I can fix it a little bit later

**Review by elena-lyulina — APPROVED** [review]

**Pull request merged**